lib/common/xml.c

```diff
@@ -510,24 +512,34 @@ find_xml_node(const xmlNode *root, const char *search_path, gboolean must_find)
  * \param[in] parent     XML element to search
  * \param[in] node_name  If not NULL, only match children of this type
  * \param[in] attr_n     If not NULL, only match children with an attribute
- *                       of this name and a value of \p attr_v
- * \param[in] attr_v     If \p attr_n and this are not NULL, only match children
- *                       with an attribute named \p attr_n and this value
+ *                       of this name.
+ * \param[in] ignore_v   If true, match children with an attribute of \p attr_n
+ *                       regardless of the value of \p attr_v
+ * \param[in] attr_v     If \p and this are not NULL and \p ignore_v is false,
+ *                       only match children with an attribute named \p attr_n
+ *                       and this value
  *
  * \return Matching XML child element, or NULL if none found
  */
 xmlNode *
 pcmk__xe_match(const xmlNode *parent, const char *node_name,
-               const char *attr_n, const char *attr_v)
+               const char *attr_n, bool ignore_v, const char *attr_v)
```

Comment:
Looking at the existing callers, attr_n and attr_v are always either both NULL or both non-NULL. So we could do this without adding another argument by using attr_v==NULL for it.

Reply:
Fixed.